service/host.go

```diff
@@ -184,32 +174,24 @@ func (s *manager) shutdown(err error, reason string, exitCode *int) (bool, error
 	return true, err
 }
 
-// addModules adds the given modules to a service manager
-func (s *manager) addModules(modules ...ModuleCreateFunc) error {
-	for _, mcf := range modules {
-		mi := ModuleCreateInfo{
-			Host:  s,
-			Items: make(map[string]interface{}),
-		}
-
-		mods, err := mcf(mi)
-		if err != nil {
-			return err
-		}
-
-		if !s.supportsRole(mi.Roles...) {
-			ulog.Logger(_simpleCtx).Info(
-				"module will not be added due to selected roles",
-				"roles", mi.Roles,
-			)
-		}
-
-		for _, mod := range mods {
-			err = s.addModule(mod)
-		}
+func (s *manager) addModule(name string, module ModuleCreateFunc, options ...ModuleOption) error {
```

Comment:
Second part of my question  - Can you update PR description with a list of main changes and why.